### PR TITLE
feat(lambda): official lambda build docker images

### LINKED ISSUE #9205: [lambda] Use the official lambda build docker images
The Lambda team has released official docker images for Lambda builds:

https://github.com/awslabs/aws-sam-cli/tree/v1.0.0/build-image-src

### Use Case

Maintain compatibility through the official images.

### Proposed Solution

The default bundling image for lambda is currently [lambci/docker-lambda](https://github.com/lambci/docker-lambda) and I think it should be moved to the new official images.

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by jogold**
Use the official lambda build docker images instead of the ones provided
by https://github.com/lambci/docker-lambda.

BREAKING CHANGE: the `bundlingDockerImage` prop of a `Runtime` now points to the AWS SAM build image (`amazon/aws-sam-cli-build-image-<runtime>`) instead of the LambCI build image (`lambci/lambda:build-<runtime>`)

Closes #9205

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by adamelmore** [comment]
@jogold do you mind to update the `aws-lambda-python` README [here](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-lambda-python/README.md#module-dependencies) as well? Or I can follow-up with a PR otherwise.

**Comment by jogold** [comment]
> @jogold do you mind to update the `aws-lambda-python` README [here](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-lambda-python/README.md?rgh-link-date=2020-07-23T06%3A43%3A10Z#module-dependencies) as well? Or I can follow-up with a PR otherwise.

done

File: packages/@aws-cdk/aws-lambda-nodejs/README.md [hunk 1]
```diff
@@ -102,4 +102,4 @@ new lambda.NodejsFunction(this, 'my-handler', {
 The modules listed in `nodeModules` must be present in the `package.json`'s dependencies. The
 same version will be used for installation. If a lock file is detected (`package-lock.json` or
 `yarn.lock`) it will be used along with the right installer (`npm` or `yarn`). The modules are
-installed in a [Lambda compatible Docker container](https://github.com/lambci/docker-lambda).
+installed in a Lambda compatible Docker container.
```
Comment (eladb) [inline packages/@aws-cdk/aws-lambda-nodejs/README.md:105]: Worth leaving a link here to the official source on docker hub

Comment (jogold) [reply packages/@aws-cdk/aws-lambda-nodejs/README.md]: It's one docker hub repo per runtime so I linked to the "default" runtime (in aws-lambda-nodejs and aws-lambda-python).

**Pull request merged**